### ISSUE TITLE
Environment overrides for `network` are overwriting base definitions.

### ISSUE THREAD

**Issue by matthewhembree** [aws/copilot-cli#3346]
I'm am seeing this with `network:`. I am not sure if this is happening with other maps.

In comparison, this isn't merging in the way that I would expect helm would merge: Maps/keys are merged at all levels. Values/arrays are overwritten.

Is everything getting overridden at the map of `network:`?

This doesn't place the task in the private subnet:
```yaml
network:
  vpc:
    placement: private

environments:
  dev:
    network:
      vpc:
        security_groups:
          - sg-1234567890
```

This does place the task in the private subnet:
```yaml
environments:
  dev:
    network:
      vpc:
        placement: private
        security_groups:
          - sg-1234567890
```

edit: Thanks!

**Comment by Lou1415926** [comment]
Hi @matthewhembree ! Thank you very much for reporting the bug. The fix PR is out: #3352!

**Comment by efekarakus** [comment]
Hi folks! The fix is now released https://github.com/aws/copilot-cli/releases/tag/v1.16.0 🥳 

See announcement: https://aws.github.io/copilot-cli/blogs/release-v116/
